### PR TITLE
[0.1] Support input field in calls.

### PR DESCRIPTION
**Pull request by yarkinwho**
For #83

### CODE REVIEW

**Review by arhag — APPROVED** [review]

File: silkworm/silkrpc/commands/eth_api.cpp [hunk 1]
```diff
@@ -1245,14 +1248,14 @@ awaitable<void> EthereumRpcApi::handle_eth_create_access_list(const nlohmann::js
...
-    auto call = params[0].get<Call>();
-    const auto block_number_or_hash = params[1].get<BlockNumberOrHash>();
-
-    SILK_DEBUG << "call: " << call << " block_number_or_hash: " << block_number_or_hash;
-
+    
     auto tx = co_await database_->begin();
 
     try {
+        auto call = params[0].get<Call>();
```
Comment (arhag) [inline silkworm/silkrpc/commands/eth_api.cpp:1255]: `const auto bundles = params[0].get<Bundles>();` within `EthereumRpcApi::handle_eth_call_many` below could also throw now. That too would need to be put into the try catch.

File: silkworm/silkrpc/commands/debug_api.cpp [hunk 2]
```diff
@@ -329,27 +321,46 @@ awaitable<void> DebugRpcApi::handle_debug_trace_call(const nlohmann::json& reque
     auto tx = co_await database_->begin();
 
     try {
-        ethdb::TransactionDatabase tx_database{*tx};
-        ethdb::kv::CachedDatabase cached_database{block_number_or_hash, *tx, *state_cache_};
+        const auto call = params[0].get<Call>();
```
Comment (arhag) [inline silkworm/silkrpc/commands/debug_api.cpp:324]: `const auto bundles = params[0].get<Bundles>();` within `DebugRpcApi::handle_debug_trace_call_many` below could also throw now. That too would need to be put into the try catch.

**Review by arhag — CHANGES_REQUESTED** [review]

File: silkworm/silkrpc/commands/trace_api.cpp [hunk 3]
```diff
@@ -44,15 +44,15 @@ boost::asio::awaitable<void> TraceRpcApi::handle_trace_call(const nlohmann::json
...
-    const auto call = params[0].get<Call>();
-    const auto config = params[1].get<trace::TraceConfig>();
-    const auto block_number_or_hash = params[2].get<BlockNumberOrHash>();
-
-    SILK_INFO << "call: " << call << " block_number_or_hash: " << block_number_or_hash << " config: " << config;
-
     auto tx = co_await database_->begin();
 
     try {
+        const auto call = params[0].get<Call>();
```
[the remark this answers is not in the document]
Comment (yarkinwho) [reply silkworm/silkrpc/commands/trace_api.cpp:50]: fixed

File: silkworm/silkrpc/commands/eth_api.cpp (hunk 1 above)
Comment (yarkinwho) [reply silkworm/silkrpc/commands/eth_api.cpp:1257]: fixed

File: silkworm/silkrpc/commands/debug_api.cpp (hunk 2 above)
Comment (yarkinwho) [reply silkworm/silkrpc/commands/debug_api.cpp:324]: fixed

**Comment by yarkinwho** [comment]
We decided we do not need to catch deserialization errors in handler

**Review by arhag — APPROVED** [review]

**Pull request merged**